pyvrp/tests/test_ProblemData.py

```diff
@@ -273,3 +273,41 @@ def test_matrices_are_not_copies():
     dur2 = data.duration_matrix()
     assert_(not dur1.flags["OWNDATA"])
     assert_(dur1.base is dur2.base)
+
+
+@pytest.mark.parametrize(
+    ("capacity", "num_available"),
+    [
+        (0, 0),  # num_available must be positive
+        (-1, 0),  # capacity cannot be negative
+        (-100, 0),  # this is just wrong
+    ],
+)
+def test_vehicle_type_raises_invalid_data(capacity: int, num_available: int):
+    """
+    Tests that the vehicle type constructor raises when given invalid
+    arguments.
+    """
+    with assert_raises(ValueError):
+        VehicleType(capacity, num_available)
+
```

Comment:
Yes, but there are no restrictions on the `fixed_cost`, so there aren't any invalid argument values for it. We could add the argument here but that's not really adding to the point of testing that the constructor raises for arguments that don't make sense.

Reply:
Oh, I didn't see that there were no restrictions to `fixed_cost`. It's a bit odd to allow for negative costs. Don't we disallow negative duration/distance/prizes for similar reasons?

Reply:
I can imagine it might help incentivise using a particular vehicle type, but that could also be done by making the other types more expensive.

> Do we want to impose non-negative costs only? I can't really think of an example where it's useful to have negative costs.

I don't know. I do not want to disallow too many things just because we could not think of a clear use case. I don't want our lack of imagination to limit other people, but I guess they could just open an issue if they run into problems because of it.